### PR TITLE
Add documentation for using uv with PyTorch

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Now that we have all the pieces in place, this PR adds some dedicated documentation to enable a variety of PyTorch setups.

This PR is downstream of #6523 and builds on the content in there; #6523 will merge first, and this PR will follow.

### CODE REVIEW

File: docs/guides/integration/pytorch.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+requires-python = ">=3.12"
+dependencies = [
+  "torch>=2.5.1",
+  "torchvision>=0.20.1",
+]
+```
+
+!!! tip "Supported Python versions"
+
+At time of writing, PyTorch does not yet publish wheels for Python 3.13; as such projects with
```
Comment (zanieb) [inline docs/guides/integration/pytorch.md:44]: Needs indent

File: docs/guides/integration/pytorch.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    name = "pytorch-rocm"
+    url = "https://download.pytorch.org/whl/rocm6.2"
+    explicit = true
+    ```
+
+We recommend the use of `explicit = true` to ensure that the index is _only_ used for `torch`,
+`torchvision`, and other PyTorch-related packages, as opposed to generic dependencies like `jinja2`,
+which should continue to be sourced from the default index (PyPI).
+
+Next, we'll update the `pyproject.toml` to point `torch` and `torchvision` to the desired index:
```
Comment (zanieb) [inline docs/guides/integration/pytorch.md:108]: Nit: We don't really use the "we will" voice

```suggestion
Next, update the `pyproject.toml` to point `torch` and `torchvision` to the desired index:
```

**Review by zanieb — APPROVED** [review]

**Comment by UnknownPlatypus** [comment]
This is awesome, thanks a lot :raised_hands:

File: mkdocs.template.yml [hunk 3]
```diff
@@ -115,6 +115,7 @@ nav:
       - GitHub Actions: guides/integration/github.md
       - GitLab CI/CD: guides/integration/gitlab.md
       - Pre-commit: guides/integration/pre-commit.md
+      - PyTorch: guides/integration/pytorch.md
```
Comment (cthoyt) [inline mkdocs.template.yml:118]: Looks like this is a duplicate, see below the fastapi line

Comment (YoniChechik) [reply mkdocs.template.yml:118]: fixed in #9220

**Pull request merged**